### ISSUE TITLE
Can't verify email

### ISSUE THREAD

**Issue by ProLoser** [airpair/airpair-com-v1-issue-tracker#59]
It says the hash is invalid, this is the emailed hash:

https://www.airpair.com/me?verify=$2a$08$hTgcLeyCjbK4919KHmaM0OmtcaI17/QI1FMpKuGK625S8P/JeqHC.

as you can see there are `/` characters in this hash and they are being converted to %7 or something when I click on the link. My guess is that this is the reason I can't verify.

**Comment by jkresner** [comment]
I think it's the . at the end. Is it still not working with the . included?

**Comment by ProLoser** [comment]
Works. Thanks.

On Tue, Jan 6, 2015 at 10:56 PM, Jonathon Kresner <notifications@github.com>
wrote:

> I think it's the . at the end. Is it still not working with the . included?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/airpair/airpair-com-v1-issue-tracker/issues/59#issuecomment-68985755>
> .
>



-- 
Dean J Sofer
DeanSofer@gmail.com

BS Computer Information Systems
California State Polytechnic University, Pomona
Telephone: 714-900-2254
Website: www.DeanSofer.com
Github: github.com/ProLoser

**Comment by eazyigz** [comment]
I am having the same problem!  No dots at the end.

**Comment by jkresner** [comment]
@eazyigz, database says your email is valid. I'm sure if you refresh the page - or at worst logout and back in again you'll be good to go. I'm aware of this issue and have it in the backlog for this weekend alongs with accounting for the dot problem.

**Comment by jkresner** [comment]
@ProLoser, @IgorGanapolsky I have deployed a rather comprehensive refactor of our session and auth architecture. The . bug should be gone another with any other permutations of emailVerification issues. I have credit both your accounts $10. Thanks for reporting the issue.

**Comment by bf4** [comment]
I'm happy to open a new issue:

This is happening to me to. I signed in via github, when to account, clicked reset password, clicked link, and got the 'invalid hash' error

**Comment by jkresner** [comment]
@bf4 was there a . character at the end of your hash? Can you forward the email you got to team@ airpair

**Comment by bf4** [comment]
no `.` at the end... email sent

**Comment by jkresner** [comment]
My hunch is that your primary email had a +github before the @ sign, and
it's possible that is causing an issue. I've just sent you a new reset and
updated your email on your user record. Let's see what happens.

Jonathon
ᐧ

On Mon, Nov 23, 2015 at 2:09 PM, Benjamin Fleischer <
notifications@github.com> wrote:

> no . at the end... email sent
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/airpair/airpair-com-v1-issue-tracker/issues/59#issuecomment-158839716>
> .
>



-- 
*Jonathon Kresner*
*Co-founder, CEO*
@AirPair <http://twitter.com/airpair> | *airpair.com
<http://www.airpair.com>*
@hackerpreneur <http://twitter.com/hackerpreneur>

**Comment by bf4** [comment]
That worked, but my GitHub auth is now gone, as is my request, and it seems to think I'm authorized for google oauth login, which isn't what I wanted.

I can still log in with via GitHub which appears to be a different account.

So, now that I'm logged in via GitHub, I clicked the Google oauth thing and am now logged in there with my work google account, but the password reset still goes to my personal gmail account linked to github.

But, I figured it out.  The password reset link had `email=handle%20github@gmail.com` but should have been `handle%2Bgithub@gmail.com`. When I manually changed it to `CGI.escape('+') #=> '%2B'` I was able to change the password
